reana_client/cli/files.py

```diff
@@ -66,6 +67,14 @@ def files_group(ctx):
     default=None,
     help="Get URLs of output files.",
 )
+@click.option(
+    "--filter",
+    "filters",
+    multiple=True,
```

Comment:
The multiple `--filter` option does not seem to work for me:

```console
$ reana-client ls -w helloworld-cwl-kubernetes --filter name=inputs.json 
NAME          SIZE   LAST-MODIFIED      
inputs.json   145    2021-06-14T14:33:28

$ $ reana-client ls -w helloworld-cwl-kubernetes --filter last-modified=2021-06-14T14:33:28
NAME            SIZE   LAST-MODIFIED      
workflow.json   1287   2021-06-14T14:33:28
inputs.json     145    2021-06-14T14:33:28

$ reana-client ls -w helloworld-cwl-kubernetes --filter last-modified=2021-06-14T14:33:28 --filter name=inputs.json
NAME   SIZE   LAST-MODIFIED
```

The last command, filtering on both file name and modification date, should have found `inputs.json`, but it did not find anything.

Can you please fix and add a test case?

Reply:
If we want to apply multiple filters, then this would be like this:

```
reana-client ls -w workflow.1 --filter last-modified=2021-06-14T10:20:17,name=inputs
```
<img width="889" alt="Screenshot 2021-06-14 at 5 01 59 PM" src="https://user-images.githubusercontent.com/24358501/121914108-4420ef00-cd32-11eb-9c6b-f82b27a55aa2.png">

Reply:
In that case we should not allow `--filter` to be `multiple=True` so that there wouldn't be a mix-up. 

(FWIW I think `--filter name1=value1 --filter name2=value2` might be more readable.)

There still seem to be a functional problem in applying two filters:

```console
$ reana-client ls -w helloworld-cwl-kubernetes --filter last-modified=39 
NAME            SIZE   LAST-MODIFIED      
workflow.json   1287   2021-06-14T17:57:39
inputs.json     145    2021-06-14T17:57:39 
$ reana-client ls -w helloworld-cwl-kubernetes --filter last-modified=39,name=i
NAME                                                                        SIZE   LAST-MODIFIED      
workflow.json                                                               1287   2021-06-14T17:57:39
inputs.json                                                                 145    2021-06-14T17:57:39
outputs/greetings.txt                                                       34     2021-06-14T17:57:42
cwl/docker_stagedir/stg42f6fb33-f98f-480f-a78c-ba5d2c1e9c3d/helloworld.py   3253   2021-06-14T17:57:27
cwl/docker_stagedir/stg317909e8-405c-4f42-b5d6-b9dab153a39f/names.txt       20     2021-06-14T17:57:27
cwl/docker_outdir/results/greetings.txt                                     34     2021-06-14T17:57:42
```

The filters are not applied one after each other, i.e. not in an AND manner but rather in an OR manner.  Whilst the most typical expectation would be Unix pipe: `cat file.txt | grep foo | grep bar` which is applying two filters, one for foo and one for bar, as it were.  A good analogy with our `--filter foo -- filter bar`.

We should probably fix this, also for the `list` etc commands?

```console
$ reana-client list  --filter name=cwl,name=yad
NAME                        RUN_NUMBER   CREATED               STARTED               ENDED                 STATUS  
helloworld-cwl-kubernetes   1            2021-06-14T17:57:26   2021-06-14T17:57:39   2021-06-14T17:57:47   finished
$ reana-client list  --filter name=cwl --filter name=yad
NAME   RUN_NUMBER   CREATED   STARTED   ENDED   STATUS
```

BTW note the difference:

```console
$ reana-client list  --filter name=cwl,name=yad
NAME                        RUN_NUMBER   CREATED               STARTED               ENDED                 STATUS  
helloworld-cwl-kubernetes   1            2021-06-14T17:57:26   2021-06-14T17:57:39   2021-06-14T17:57:47   finished
$ reana-client list  --filter name=yad,name=cwl         
NAME                           RUN_NUMBER   CREATED               STARTED               ENDED                 STATUS  
helloworld-yadage-kubernetes   1            2021-06-14T17:57:53   2021-06-14T17:58:07   2021-06-14T17:58:29   finished
```

Reply:
Here is a pseudo-code of how I picture filters should work:

```python
for item in items:
   for filter in filters:
      if filter.wanted_value not in item.get(filter.wanted_key, ""):
         break
      output.append(item) # all filters matched, so retain the item
return output
```

Reply:
@tiborsimko Added the changes in `ls` to respect `multiple=True`.  Included it in tests as well.

You can test with following commands to verify if the working is as expected now.

```
reana-client ls -w workflow.1 --filter last-modified=20 --filter name=hello
```